WCF/Shared/Implementation/ClientTelemetryChannelBase.cs

```diff
@@ -0,0 +1,361 @@
+        }
+
+        public abstract EndpointAddress RemoteAddress { get; }
+
+        public event EventHandler Closed;
+        public event EventHandler Closing;
+        public event EventHandler Faulted;
+        public event EventHandler Opened;
+        public event EventHandler Opening;
+
+        public ClientTelemetryChannelBase(IChannelManager channelManager, IChannel channel)
+        {
+            if ( channelManager == null )
+            {
+                throw new ArgumentNullException(nameof(channelManager));
+            }
+            if ( channel == null )
+            {
+                throw new ArgumentNullException(nameof(channel));
+            }
+            this.ChannelManager = channelManager;
+            this.InnerChannel = channel;
+        }
+
+        public void Open()
+        {
+            Open(ChannelManager.OpenTimeout);
+        }
+
+        public void Open(TimeSpan timeout)
+        {
+            WcfClientEventSource.Log.ChannelCalled(GetType().FullName, nameof(Open));
+            HookChannelEvents();
+            var telemetry = this.StartOpenTelemetry(nameof(Open));
+            try
+            {
+                InnerChannel.Open(timeout);
+                this.StopOpenTelemetry(telemetry, null, nameof(Open));
+            } catch ( Exception ex )
+            {
+                this.StopOpenTelemetry(telemetry, ex, nameof(Open));
+                throw;
+            }
+        }
+
+        public void Close()
+        {
+            Close(ChannelManager.CloseTimeout);
+        }
+
+        public void Close(TimeSpan timeout)
+        {
+            WcfClientEventSource.Log.ChannelCalled(GetType().FullName, nameof(Close));
+            try
+            {
+                InnerChannel.Close(timeout);
+            } finally
+            {
+                Dispose(true);
+            }
+        }
+
+        public void Abort()
+        {
+            WcfClientEventSource.Log.ChannelCalled(GetType().FullName, nameof(Abort));
+            try
+            {
+                InnerChannel.Abort();
+            } finally
+            {
+                Dispose(true);
+            }
+        }
+
+        public T GetProperty<T>() where T : class
+        {
+            return InnerChannel.GetProperty<T>();
+        }
+
+        //
+        // Async Methods
+        //
+        public IAsyncResult BeginOpen(AsyncCallback callback, object state)
+        {
+            return BeginOpen(this.ChannelManager.OpenTimeout, callback, state);
+        }
+
+        public IAsyncResult BeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            WcfClientEventSource.Log.ChannelCalled(GetType().FullName, nameof(BeginOpen));
+            HookChannelEvents();
+            var telemetry = StartOpenTelemetry(nameof(BeginOpen));
+            try
+            {
+                return new OpenAsyncResult(InnerChannel, timeout, this.OpenCompleted, callback, state, telemetry);
```

Comment:
Not when the channel is being opened. At that point, messages *could* be exchanged with the service, but they would be produced by the channels underneath us.

For example, if you're using WSHttpBinding with Message-based security and `negotiateServiceCredential` is enabled, the security channel will exchange infrastructure messages with the remote endpoint (doing 1 or more HTTP requests) but those will get handled by the WCF infrastructure and not make it to the service (the server side tracking won't see those requests, either).

We could potentially inject a second channel at the bottom of the stack (right before the encoding element) to do it, but not sure if it's worth it or not.